### PR TITLE
device: esp32: Convert clock control to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by galak**
Replace device_get_binding with DEVICE_DT_GET for getting access
to the clock controller device.

### CODE REVIEW

**Comment by galak** [comment]
blocked on #32188

File: drivers/i2c/i2c_esp32.c [hunk 1]
```diff
@@ -671,9 +669,6 @@ static int i2c_esp32_init(const struct device *dev)
 	const struct i2c_esp32_config *config = dev->config;
 	struct i2c_esp32_data *data = dev->data;
 	uint32_t bitrate_cfg = i2c_map_dt_bitrate(config->bitrate);
-	data->clock_dev = device_get_binding(config->clock_name);
```
Comment (pabigot) [inline drivers/i2c/i2c_esp32.c:674]: There should be an ASSERT that `config->clock_dev` is ready, either here or in the functions that use it.

File: drivers/serial/uart_esp32.c [hunk 2]
```diff
@@ -317,12 +316,6 @@ static int uart_esp32_configure(const struct device *dev,
 
 static int uart_esp32_init(const struct device *dev)
 {
-	struct uart_esp32_data *data = DEV_DATA(dev);
-
-	data->clock_dev = device_get_binding(DEV_CFG(dev)->clock_name);
-
-	__ASSERT_NO_MSG(data->clock_dev);
-
 	uart_esp32_configure(dev, &DEV_DATA(dev)->uart_config);
```
Comment (pabigot) [inline drivers/serial/uart_esp32.c:319]: here too.

**Review by pabigot — CHANGES_REQUESTED** [review]

**Review by pabigot — APPROVED** [review]
Error handling ok pending switch to #32272

**Review by dcpleung — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**